workspaces/tech-radar/plugins/tech-radar/config.d.ts

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright 2020 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+export interface Config {
+  techRadar?: {
+    /**
+     * Optional URL pointing to the definition file for the tech radar entries. If both `url` & `entries` are specified, the `url` will be used.
+     * @visibility frontend
+     */
+    url?: string;
+    /**
+     * Optional representation of the tech radar entries. Will only be used if `url` is not provided.
+     * @visibility frontend
+     */
+    graphData?: {
```

Comment:
This section is clunky/slightly sub-optimal to have the type definition functionally duplicated from the RadarRing/RadarQuandrant/RadarEntry types in ./api.ts - but 1) the types are pretty stables and 2) typing it out fully in the config syntax provides better readability for users interacting with the entry via ConfigManager in Portal, so thought it was worth the dupe.

Reply:
it looks a bit complicated to use app config for tech radar, given how deep structured the data is. From a devex perspective probably it would be easier to use the url approach 🤔

Reply:
I do agree / think the URL is a nicer approach - but I think the ability to do it directly in the config is a decent 3rd tier option & it doesn't hurt to have it implemented. I can remove it if you feel strongly, but I'd prefer to keep it in